### PR TITLE
Document Pokemon related structs and functions

### PR DESCRIPTION
**Pull request by AsparagusEduardo**
## Renamed Files
- `include/struct_decls/struct_02073C74_sub1_decl.h` →`include/struct_decls/box_pokemon.h`
- `include/struct_defs/struct_02073C74_sub1.h` → `include/struct_defs/box_pokemon.h`
- `include/struct_defs/struct_02075454.h` → `include/struct_defs/pokemon_substruct_0.h`
- `include/struct_defs/struct_02075454_1.h` → `include/struct_defs/pokemon_substruct_1.h`
- `include/struct_defs/struct_02075454_2.h` → `include/struct_defs/pokemon_substruct_2.h`
- `include/struct_defs/struct_02075454_3.h` → `include/struct_defs/pokemon_substruct_3.h`
- `include/struct_defs/struct_02073C74_sub2_t.h` → `include/struct_defs/party_pokemon.h`
- `include/struct_decls/struct_02073C74_sub2_decl.h` → `include/struct_decls/party_pokemon.h`
- `include/overlay097/struct_ov97_0223685C.h` → `include/overlay097/box_pokemon_gba.h`

## Deleted Unneeded Files
- `include/struct_decls/struct_02075454_decl.h`
- `include/struct_decls/struct_02075454_1_decl.h`
- `include/struct_decls/struct_02075454_2_decl.h`
- `include/struct_decls/struct_02075454_3_decl.h`
- `include/struct_decls/struct_02073C74_decl.h`

## Renamed Structs and Fields
- `UnkStruct_02073C74` → `Pokemon`
	- `unk_00` → `box`
	- `unk_08` → `party`
- `UnkStruct_02073C74_sub1` → `BoxPokemon`
- `UnkStruct_02073C74_sub2` → `PartyPokemon`
- `UnkStruct_02075454` → `PokemonSubstruct0`
	- `unk_00` → `species`
	- `unk_02` → `item`
	- `unk_0C` → `friendship`
- `UnkStruct_02075454_1` → `PokemonSubstruct1`
	- `unk_18_0` → `fatefulEncounter`
	- `unk_18_3` → `form`
- `UnkStruct_02075454_2` → `PokemonSubstruct2`
- `UnkStruct_02075454_3` → `PokemonSubstruct3`

#### GBA Structs/Fields
- `UnkStruct_ov97_0223685C` → `BoxPokemonGBA`
	- `unk_00` → `personality`
	- `unk_04` → `otId`
	- `unk_08` → `nickname`
	- `unk_12` → `language`
	- `unk_13_0` → `isBadEgg`
	- `unk_13_1` → `hasSpecies`
	- `unk_13_2` → `isEgg`
	- `unk_13_3/4` → `unused`
	- `unk_14` → `otName`
	- `unk_1B` → `markings`
	- `unk_1C` → `checksum`
	- `unk_1E` → `unknown`
	- `unk_20` → `secure`
- `UnkStruct_ov97_02236380_sub1` → `PokemonGBASubstruct0`
	- `unk_00` → `species`
	- `unk_02` → `heldItem`
	- `unk_04` → `experience`
	- `unk_08` → `ppBonuses`
	- `unk_09` → `friendship`
	- `unk_0A` → `padding`
- `UnkStruct_ov97_02236380_sub2` → `PokemonGBASubstruct1`
	- `unk_00` → `moves`
	- `unk_08` → `pp`
- `UnkStruct_ov97_02236380_sub3` → `PokemonGBASubstruct2`
	- `unk_00` → `hpEV`
	- `unk_01` → `attackEV`
	- `unk_02` → `defenseEV`
	- `unk_03` → `speedEV`
	- `unk_04` → `spAttackEV`
	- `unk_05` → `spDefenseEV`
	- `unk_06` → `cool`
	- `unk_07` → `beauty`
	- `unk_08` → `cute`
	- `unk_09` → `smart`
	- `unk_0A` → `tough`
	- `unk_0B` → `sheen`
- `UnkStruct_ov97_02236380_sub4` → `PokemonGBASubstruct3`
	- `unk_00_0` → `pokerus`
	- `unk_00_8` → `metLocation`
	- `unk_00_16` → `metLevel`
	- `unk_00_23` → `metGame`
	- `unk_00_27` → `pokeball`
	- `unk_00_31` → `otGender`
	- `unk_04_0` → `hpIV`
	- `unk_04_5` → `attackIV`
	- `unk_04_10` → `defenseIV`
	- `unk_04_15` → `speedIV`
	- `unk_04_20` → `spAttackIV`
	- `unk_04_25` → `spDefenseIV`
	- `unk_04_30` → `isEgg`
	- `unk_04_31` → `abilityNum`
	- `unk_08_0` → `coolRibbon`
	- `unk_08_3` → `beautyRibbon`
	- `unk_08_6` → `cuteRibbon`
	- `unk_08_9` → `smartRibbon`
	- `unk_08_12` → `toughRibbon`
	- `unk_08_15` → `championRibbon`
	- `unk_08_16` → `winningRibbon`
	- `unk_08_17` → `victoryRibbon`
	- `unk_08_18` → `artistRibbon`
	- `unk_08_19` → `effortRibbon`
	- `unk_08_20` → `marineRibbon`
	- `unk_08_21` → `landRibbon`
	- `unk_08_22` → `skyRibbon`
	- `unk_08_23` → `countryRibbon`
	- `unk_08_24` → `nationalRibbon`
	- `unk_08_25` → `earthRibbon`
	- `unk_08_26` → `worldRibbon`
	- `unk_08_27` → `unusedRibbons`
	- `unk_08_31` → `modernFatefulEncounter`
- `UnkStruct_ov97_02236340` → `PokemonStorageGBA`
	- `unk_00` → `currentBox`
	- `unk_04` → `boxes`
	- `unk_8344` → `boxNames`
	- `unk_83C2` → `boxWallpapers`

#### `src/overlay012/is_form_symmetrical.c`
- `ov12_022384F0` → `IsFormSymmetrical`
#### `src/unk_02073C2C.c`
- `sub_02073C70` → `PokemonStructSize`
- `sub_02073C74` → `AllocMonZeroed`
#### `src/overlay097/ov97_02233B18.c`
- `ov97_02233B18` → `IsGBASpeciesInvalid`

#### `src/unk_02073C2C.c`
- `sub_02073C2C` → `ZeroMonData`
- `sub_02073C54` → `ZeroBoxMonData`
- `sub_02074470` → `GetMonData`
- `sub_02075BFC` → `GetNatureFromPersonality`
- `sub_02077C20` → `SetRotomForm`
- `sub_02077ACC` → `SetShayminForm`
- `sub_02077ADC` → `SetBoxShayminForm`
- `sub_020761E8` → `SanitizeFormId`
- `sub_02074C60` → `SetBoxMonData`
#### `src/unk_0205DFC4.c`
- `sub_0205E1F8` → `HasAllLegendaryTitansInParty`

#### `src/overlay097/ov97_02236380.c`
- `ov97_0223685C` → `GBADecryptBoxMon`
- `ov97_0223687C` → `GBAEncryptBoxMon`
- `ov97_02236924` → `GetGBABoxMonData`
- `ov97_02236CA4` → `SetGBABoxMonData`
- `ov97_02236E90` → `BoxMonGBAToBoxMon`

## New Enums and Defines
- Enums:
	- `MON_DATA_x`
	- `GBA_MON_DATA_x`
- Defines:
	- `GBA_SPECIES_x`

AsparagusEduardo#6051

### CODE REVIEW

**Comment by AsparagusEduardo** [comment]
Might as well mark this as ready, as I haven't added new commits for over a month.

File: src/overlay012/is_form_symmetrical.c [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    case SPECIES_BUDEW:
+    case SPECIES_ROSERADE:
+    case SPECIES_MAGMORTAR:
+    case SPECIES_TORTERRA:
+    case SPECIES_TOGEKISS:
+    case SPECIES_KINGLER:
+        return FALSE;
+    case SPECIES_SHAYMIN:
+        if (formId == 1)
+            return FALSE;
```
Comment (luckytyphlosion) [inline src/overlay012/is_form_symmetrical.c:28]: braces even on one statement

File: src/overlay097/ov97_02233B18.c [hunk 2]
```diff
@@ -232,25 +235,18 @@ static int Unk_ov97_0223EAB8[] = {
...
-
-    return 1;
+    if (speciesGBA <= SPECIES_NONE)
+        return TRUE;
+    else if (speciesGBA <= SPECIES_CELEBI)
+        return FALSE;
+    else if (speciesGBA < GBA_SPECIES_TREECKO)
+        return TRUE;
+    else if (speciesGBA <= GBA_SPECIES_CHIMECHO)
+        return FALSE;
```
Comment (luckytyphlosion) [inline src/overlay097/ov97_02233B18.c:247]: braces even on single statement (do this for all instances)

File: src/unk_02073C2C.c [hunk 3]
```diff
@@ -111,211 +113,207 @@ static const s8 Unk_020F0695[][5] = {
...
-UnkStruct_02073C74 * sub_02073C74 (u32 param0)
+Pokemon * AllocMonZeroed (u32 param0)
 {
-    UnkStruct_02073C74 * v0;
-
-    v0 = Heap_AllocFromHeap(param0, sizeof(UnkStruct_02073C74));
-    sub_02073C2C(v0);
-
-    return v0;
+    Pokemon * mon = Heap_AllocFromHeap(param0, sizeof(Pokemon));
```
Comment (luckytyphlosion) [inline src/unk_02073C2C.c:272]: Not sure but maybe CallocMon? This diverges from gen 3 naming tho

File: src/overlay097/ov97_02236380.c [hunk 4]
```diff
@@ -960,333 +959,333 @@ static u16 ov97_0223689C (UnkStruct_ov97_0223685C * param0)
...
 
-        if (v1 == 0) {
+        if (retVal == SPECIES_NONE)
             break;
-        }
 
-        if ((v5->unk_04_30) || (param0->unk_13_0)) {
-            v1 = 412;
-        }
+        if ((substruct3->isEgg) || (boxMonGBA->isBadEgg))
```
Comment (luckytyphlosion) [inline src/overlay097/ov97_02236380.c:1204]: Get rid of redundant parens

File: src/overlay097/ov97_02236380.c [hunk 5]
```diff
@@ -960,333 +959,333 @@ static u16 ov97_0223689C (UnkStruct_ov97_0223685C * param0)
...
-            v5->unk_04_30 = 1;
+    u32 retVal = 0;
+    PokemonGBASubstruct0 * substruct0 = 0;
+    PokemonGBASubstruct1 * substruct1 = 0;
+    PokemonGBASubstruct2 * substruct2 = 0;
+    PokemonGBASubstruct3 * substruct3 = 0;
+    u16 checksum;
+
+    if (field > GBA_MON_DATA_ENCRYPT_SEPARATOR)
+    {
```
Comment (luckytyphlosion) [inline src/overlay097/ov97_02236380.c:972]: brace same line as if (and while/for/do) do for all instances

File: include/overlay097/box_pokemon_gba.h [hunk 6]
```diff
@@ -0,0 +1,95 @@
+#ifndef POKEPLATINUM_BOX_POKEMON_GBA_H
+#define POKEPLATINUM_BOX_POKEMON_GBA_H
+
+typedef struct {
+    u16 species;
+    u16 heldItem;
+    u32 experience;
+    u8 ppBonuses;
+    u8 friendship;
+    u16 filler;
```
Comment (luckytyphlosion) [inline include/overlay097/box_pokemon_gba.h:10]: this should be padding not filler, but should probably stay consistent with gen 3 repo names (i.e. this is just a note, don't change it_)

**Review by luckytyphlosion — CHANGES_REQUESTED** [review]

File: include/overlay097/box_pokemon_gba.h (hunk 6 above)
Comment (AsparagusEduardo) [reply include/overlay097/box_pokemon_gba.h:10]: I mean, you're right, it's padding, so the Gen 3 repos should rename it too.

File: src/unk_02073C2C.c (hunk 3 above)
Comment (AsparagusEduardo) [reply src/unk_02073C2C.c:272]: Hmm, I feel we can find ways to improve over Gen 3's documentation by applying more proper names now and then backporting those changes to the Gen 3 repos.

**Comment by AsparagusEduardo** [comment]
Applied review changes and updated PR description.

**Pull request merged**